src/Nethermind/Nethermind.Synchronization/SnapSync/SnapProvider.cs

```diff
@@ -91,54 +90,46 @@ public AddRangeResult AddAccountRange(
         {
             if (accounts.Count == 0)
                 throw new ArgumentException("Cannot be empty.", nameof(accounts));
-            ITrieStore store = _trieStorePool.Get();
-            try
-            {
-                StateTree tree = new(store.GetTrieStore(null), _logManager);
+            StateTree tree = new(new RawScopedTrieStore(_nodeStorage, null), _logManager);
```

Comment:
Do we need to create instance per response then? Can't we have one `RawScopedTrieStore` as a field of the class?

Reply:
So maybe reuse the one with `null` address?